StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1202ElementsMustBeOrderedByAccess.cs

```diff
@@ -125,39 +145,79 @@ private static void HandleMemberList(SyntaxNodeAnalysisContext context, SyntaxLi
 
                 // if the SyntaxKind of this member (e.g. SyntaxKind.IncompleteMember) will not
                 // be handled, skip early.
-                if (!MemberNames.ContainsKey(currentSyntaxKind))
+                if (!MemberKinds.Contains(currentSyntaxKind))
                 {
                     continue;
                 }
 
-                AccessLevel currentAccessLevel;
                 var modifiers = member.GetModifiers();
-                if ((currentSyntaxKind == SyntaxKind.ConstructorDeclaration && modifiers.Any(SyntaxKind.StaticKeyword))
-                    || (currentSyntaxKind == SyntaxKind.MethodDeclaration && (member as MethodDeclarationSyntax)?.ExplicitInterfaceSpecifier != null)
-                    || (currentSyntaxKind == SyntaxKind.PropertyDeclaration && (member as PropertyDeclarationSyntax)?.ExplicitInterfaceSpecifier != null)
-                    || (currentSyntaxKind == SyntaxKind.IndexerDeclaration && (member as IndexerDeclarationSyntax)?.ExplicitInterfaceSpecifier != null))
-                {
-                    currentAccessLevel = AccessLevel.Public;
-                }
-                else
-                {
-                    currentAccessLevel = AccessLevelHelper.GetAccessLevel(modifiers);
-                    currentAccessLevel = currentAccessLevel == AccessLevel.NotSpecified ? defaultAccessLevel : currentAccessLevel;
-                }
+                AccessLevel currentAccessLevel = MemberOrderHelper.GetAccessLevelForOrdering(member, modifiers);
 
-                if (previousAccessLevel != AccessLevel.NotSpecified
-                    && currentSyntaxKind == previousSyntaxKind
-                    && currentAccessLevel > previousAccessLevel)
+                if (previousMember != null && previousAccessLevel != AccessLevel.NotSpecified)
```

Comment:
:thought_balloon: No point in doing both checks. For the first iteration, the check is always true, after the first iteration previousMember will never be null again, so the second check is unnecessary.